rdtools/system_analysis.py

```diff
@@ -0,0 +1,583 @@
+        yoy_kwargs: dict of kwargs to pass to degradation.degradation_year_on_year()
+        srr_kwargs: dict of kwargs to pass to soiling.soiling_srr()
+
+        Returns
+        -------
+        None
+        '''
+
+        self.sensor_preprocess()
+        sensor_results = {}
+
+        if 'yoy_degradation' in analyses:
+            yoy_results = self.yoy_degradation(self.sensor_aggregated_performance, **yoy_kwargs)
+            sensor_results['yoy_degradation'] = yoy_results
+
+        if 'srr_soiling' in analyses:
+            srr_results = self.srr_soiling(self.sensor_aggregated_performance,
+                                           self.sensor_aggregated_insolation,
+                                           **srr_kwargs)
+            sensor_results['srr_soiling'] = srr_results
+
+        self.results['sensor'] = sensor_results
+
+    def clearsky_analysis(self, analyses=['yoy_degradation'], yoy_kwargs={}, srr_kwargs={}):
+        '''
+        Perform entire clear-sky-based analysis workflow. Results are stored in self.results['clearsky']
+
+        Parameters
+        ---------
+        analyses: list of analyses to perform, valid entries are 'yoy_degradation' and 'srr_soiling'
+        yoy_kwargs: dict of kwargs to pass to degradation.degradation_year_on_year()
+        srr_kwargs: dict of kwargs to pass to soiling.soiling_srr()
+
+        Returns
+        -------
+        None
+        '''
+
+        self.clearsky_preprocess()
+        clearsky_results = {}
+
+        if 'yoy_degradation' in analyses:
+            yoy_results = self.yoy_degradation(self.clearsky_aggregated_performance, **yoy_kwargs)
+            clearsky_results['yoy_degradation'] = yoy_results
+
+        if 'srr_soiling' in analyses:
+            srr_results = self.srr_soiling(self.clearsky_aggregated_performance,
+                                           self.clearsky_aggregated_insolation,
+                                           **srr_kwargs)
+            clearsky_results['srr_soiling'] = srr_results
+
+        self.results['clearsky'] = clearsky_results
+
+    def plot_degradation_summary(self, result_to_plot, **kwargs):
+        '''
+        Returns a figure of a scatter plot and a histogram summarizing degradation rate analysis.
+
+        Parameters
+        ----------
+        result_to_plot: The workflow result to plot, allowed values are 'sensor' and 'clearsky' (str)
+        kwargs: Extra parameters passed to plotting.degradation_summary_plots()
+
+        Returns
+        -------
+        matplotlib figure containing two axes
+
+        '''
+
+        if result_to_plot == 'sensor':
+            results_dict = self.results['sensor']['yoy_degradation']
+            aggregated = self.sensor_aggregated_performance
+        elif result_to_plot == 'clearsky':
+            results_dict = self.results['clearsky']['yoy_degradation']
+            aggregated = self.clearsky_aggregated_performance
```

Comment:
Could add an `else` here to catch typos and throw a more informative error than `NameError: name 'results_dict' is not defined`.  The other plot functions as well